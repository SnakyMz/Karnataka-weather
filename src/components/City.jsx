import { useSelector } from 'react-redux';
import { Link, useParams } from 'react-router-dom';
import back from '../images/left-arrow.png';
import microphone from '../images/microphone.png';
import setting from '../images/settings.png';
import '../styles/City.css';

function City() {
  const { cityName } = useParams();

  const cities = useSelector((state) => state.karnataka.weather);
  const city = cities.find((data) => data.name === cityName);

  return (
    <>
      <header>
        <Link to="/"><img src={back} alt="Back" className="icons links" /></Link>
        <h4>
          {city.name}
          &apos;
          s Weather
        </h4>
        <div className="navIcons">
          <img src={microphone} alt="Microphone" className="icons" />
          <img src={setting} alt="Settings" className="icons setting" />
        </div>
      </header>

      <section className="cityImg" style={{ backgroundImage: `url(${city.img})` }}>
        <h1>{city.name}</h1>
      </section>

      <section className="weatherDetails">
        <h5 className="detailHead">Details</h5>
        <h4>
          Weather:
          &nbsp;
          {city.weather.weather[0].description}
        </h4>
        <h4>
          Temperature:
          &nbsp;
          {city.weather.main.temp}
        </h4>
        <h4>
          Humidity:
          &nbsp;
          {city.weather.main.humidity}
        </h4>
        <h4>
          Wind:
          &nbsp;
          {city.weather.wind.speed}
        </h4>
      </section>
    </>
  );
}

export default City;
