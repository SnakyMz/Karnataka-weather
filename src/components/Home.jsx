import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import { karnatakaWeather } from '../redux/karnataka/karnatakaSlice';
import forward from '../images/right-arrow.png';
import reload from '../images/reload.png';
import Nav from './Nav';
import '../styles/Home.css';

function Home() {
  const cities = useSelector((state) => state.karnataka.weather);
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(karnatakaWeather());
  }, [dispatch]);

  return (
    <>
      <header>
        <Link to="/"><img src={reload} alt="Reload" className="icons links" /></Link>
        <h4>Karnataka Weather</h4>
        <Nav />
      </header>

      <section className="titleDiv">
        <h1>Karnataka</h1>
        <p>One State, Many Worlds</p>
      </section>

      <section id="cities">
        <h5 className="citiesHead">Cities</h5>
        <div className="citiesList">
          {cities.map((city) => (
            <Link to={`/${city.name}`} key={city.name} className="links">
              <div className="city" style={{ backgroundImage: `url(${city.img})` }}>
                <img src={forward} alt="Forward" className="icons" />
                <h3 className="cityHead">{city.name}</h3>
                <p>{city.weather.main.temp}</p>
              </div>
            </Link>
          ))}
        </div>
      </section>
    </>
  );
}

export default Home;
