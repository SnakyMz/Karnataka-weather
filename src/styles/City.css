.cityImg {
  min-height: 10rem;
  width: 100%;
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  background-color: dodgerblue;
  background-blend-mode: multiply;
  text-align: right;
  display: flex;
  justify-content: right;
  align-items: center;
  padding-right: 5%;
}

.detailHead {
  padding: 0.3% 3%;
}

.weatherDetails h4 {
  padding: 2% 3%;
}

.weatherDetails h4:nth-child(odd) {
  background-color: dodgerblue;
}

.weatherDetails h4:nth-child(even) {
  background-color: deepskyblue;
}

@media screen and (min-width: 768px) {
  .cityImg {
    min-height: 30rem;
  }
}
