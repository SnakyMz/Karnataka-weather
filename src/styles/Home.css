@import url('https://fonts.googleapis.com/css2?family=Lato:wght@300;400;700&family=Martel+Sans:wght@300;400;700&display=swap');

body {
  font-family: 'Martel Sans', sans-serif;
  background-color: royalblue;
}

header {
  font-family: 'Lato', sans-serif;
}
