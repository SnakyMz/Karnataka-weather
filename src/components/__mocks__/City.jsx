import back from '../../images/left-arrow.png';
import microphone from '../../images/microphone.png';
import setting from '../../images/settings.png';
import bangalore from '../../images/bangalore.jpg';
import '../../styles/City.css';

function City() {
  const cityData = {
    name: 'Bangalore',
    img: bangalore,
    weather: {
      main: {
        temp: 25,
        humidity: 8000,
      },
      weather: [
        {
          description: 'Sunny',
        },
      ],
      wind: {
        speed: 235,
      },
    },
  };

  return (
    <>
      <header>
        <a href="../Home.jsx"><img src={back} alt="Back" className="icons links" /></a>
        <h4>
          {cityData.name}
          &apos;
          s Weather
        </h4>
        <div className="navIcons">
          <img src={microphone} alt="Microphone" className="icons" />
          <img src={setting} alt="Settings" className="icons setting" />
        </div>
      </header>

      <section className="cityImg" style={{ backgroundImage: `url(${cityData.img})` }}>
        <h1>{cityData.name}</h1>
      </section>

      <section className="weatherDetails">
        <h5 className="detailHead">Details</h5>
        <h4>
          Weather:
          &nbsp;
          {cityData.weather.weather[0].description}
        </h4>
        <h4>
          Temperature:
          &nbsp;
          {cityData.weather.main.temp}
        </h4>
        <h4>
          Humidity:
          &nbsp;
          {cityData.weather.main.humidity}
        </h4>
        <h4>
          Wind:
          &nbsp;
          {cityData.weather.wind.speed}
        </h4>
      </section>
    </>
  );
}

export default City;
