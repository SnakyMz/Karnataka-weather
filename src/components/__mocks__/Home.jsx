import forward from '../../images/right-arrow.png';
import reload from '../../images/reload.png';
import microphone from '../../images/microphone.png';
import setting from '../../images/settings.png';
import bangalore from '../../images/bangalore.jpg';
import mysore from '../../images/mysore.jpg';
import '../../styles/Home.css';

function Home() {
  const cities = [
    {
      name: 'Bangalore',
      img: bangalore,
      weather: {
        main: {
          temp: 25,
        },
      },
    },
    {
      name: 'Mysore',
      img: mysore,
      weather: {
        main: {
          temp: 30,
        },
      },
    },
  ];

  return (
    <>
      <header>
        <a href="../Home.jsx"><img src={reload} alt="Reload" className="icons links" /></a>
        <h4>Karnataka Weather</h4>
        <div className="navIcons">
          <img src={microphone} alt="Microphone" className="icons" />
          <img src={setting} alt="Settings" className="icons setting" />
        </div>
      </header>

      <section className="titleDiv">
        <h1>Karnataka</h1>
        <p>One State, Many Worlds</p>
      </section>

      <section id="cities">
        <h5 className="citiesHead">Cities</h5>
        <div className="citiesList">
          {cities.map((city) => (
            <a href="../City.jsx" key={city.name} className="links">
              <div className="city" style={{ backgroundImage: `url(${city.img})` }}>
                <img src={forward} alt="Forward" className="icons" />
                <h3 className="cityHead">{city.name}</h3>
                <p>{city.weather.main.temp}</p>
              </div>
            </a>
          ))}
        </div>
      </section>
    </>
  );
}

export default Home;
