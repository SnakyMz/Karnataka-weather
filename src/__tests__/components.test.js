import { render } from '@testing-library/react';
import Home from '../components/Home';
import City from '../components/City';

jest.mock('../components/Home');
jest.mock('../components/City');

describe('Testing components', () => {
  it('Should render Home page', () => {
    const home = render(<Home />);
    expect(home).toMatchSnapshot();
  });

  it('Should render City page', () => {
    const city = render(<City />);
    expect(city).toMatchSnapshot();
  });
});
