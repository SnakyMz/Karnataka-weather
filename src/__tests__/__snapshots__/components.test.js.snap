// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing components Should render City page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header>
        <a
          href="../Home.jsx"
        >
          <img
            alt="Back"
            class="icons links"
            src="left-arrow.png"
          />
        </a>
        <h4>
          Bangalore
          ' s Weather
        </h4>
        <div
          class="navIcons"
        >
          <img
            alt="Microphone"
            class="icons"
            src="microphone.png"
          />
          <img
            alt="Settings"
            class="icons setting"
            src="settings.png"
          />
        </div>
      </header>
      <section
        class="cityImg"
        style="background-image: url(bangalore.jpg);"
      >
        <h1>
          Bangalore
        </h1>
      </section>
      <section
        class="weatherDetails"
      >
        <h5
          class="detailHead"
        >
          Details
        </h5>
        <h4>
          Weather:  
          Sunny
        </h4>
        <h4>
          Temperature:  
          25
        </h4>
        <h4>
          Humidity:  
          8000
        </h4>
        <h4>
          Wind:  
          235
        </h4>
      </section>
    </div>
  </body>,
  "container": <div>
    <header>
      <a
        href="../Home.jsx"
      >
        <img
          alt="Back"
          class="icons links"
          src="left-arrow.png"
        />
      </a>
      <h4>
        Bangalore
        ' s Weather
      </h4>
      <div
        class="navIcons"
      >
        <img
          alt="Microphone"
          class="icons"
          src="microphone.png"
        />
        <img
          alt="Settings"
          class="icons setting"
          src="settings.png"
        />
      </div>
    </header>
    <section
      class="cityImg"
      style="background-image: url(bangalore.jpg);"
    >
      <h1>
        Bangalore
      </h1>
    </section>
    <section
      class="weatherDetails"
    >
      <h5
        class="detailHead"
      >
        Details
      </h5>
      <h4>
        Weather:  
        Sunny
      </h4>
      <h4>
        Temperature:  
        25
      </h4>
      <h4>
        Humidity:  
        8000
      </h4>
      <h4>
        Wind:  
        235
      </h4>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing components Should render Home page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header>
        <a
          href="../Home.jsx"
        >
          <img
            alt="Reload"
            class="icons links"
            src="reload.png"
          />
        </a>
        <h4>
          Karnataka Weather
        </h4>
        <div
          class="navIcons"
        >
          <img
            alt="Microphone"
            class="icons"
            src="microphone.png"
          />
          <img
            alt="Settings"
            class="icons setting"
            src="settings.png"
          />
        </div>
      </header>
      <section
        class="titleDiv"
      >
        <h1>
          Karnataka
        </h1>
        <p>
          One State, Many Worlds
        </p>
      </section>
      <section
        id="cities"
      >
        <h5
          class="citiesHead"
        >
          Cities
        </h5>
        <div
          class="citiesList"
        >
          <a
            class="links"
            href="../City.jsx"
          >
            <div
              class="city"
              style="background-image: url(bangalore.jpg);"
            >
              <img
                alt="Forward"
                class="icons"
                src="right-arrow.png"
              />
              <h3
                class="cityHead"
              >
                Bangalore
              </h3>
              <p>
                25
              </p>
            </div>
          </a>
          <a
            class="links"
            href="../City.jsx"
          >
            <div
              class="city"
              style="background-image: url(mysore.jpg);"
            >
              <img
                alt="Forward"
                class="icons"
                src="right-arrow.png"
              />
              <h3
                class="cityHead"
              >
                Mysore
              </h3>
              <p>
                30
              </p>
            </div>
          </a>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <header>
      <a
        href="../Home.jsx"
      >
        <img
          alt="Reload"
          class="icons links"
          src="reload.png"
        />
      </a>
      <h4>
        Karnataka Weather
      </h4>
      <div
        class="navIcons"
      >
        <img
          alt="Microphone"
          class="icons"
          src="microphone.png"
        />
        <img
          alt="Settings"
          class="icons setting"
          src="settings.png"
        />
      </div>
    </header>
    <section
      class="titleDiv"
    >
      <h1>
        Karnataka
      </h1>
      <p>
        One State, Many Worlds
      </p>
    </section>
    <section
      id="cities"
    >
      <h5
        class="citiesHead"
      >
        Cities
      </h5>
      <div
        class="citiesList"
      >
        <a
          class="links"
          href="../City.jsx"
        >
          <div
            class="city"
            style="background-image: url(bangalore.jpg);"
          >
            <img
              alt="Forward"
              class="icons"
              src="right-arrow.png"
            />
            <h3
              class="cityHead"
            >
              Bangalore
            </h3>
            <p>
              25
            </p>
          </div>
        </a>
        <a
          class="links"
          href="../City.jsx"
        >
          <div
            class="city"
            style="background-image: url(mysore.jpg);"
          >
            <img
              alt="Forward"
              class="icons"
              src="right-arrow.png"
            />
            <h3
              class="cityHead"
            >
              Mysore
            </h3>
            <p>
              30
            </p>
          </div>
        </a>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
